portofino-admin/src/main/resources/com/manydesigns/portofino/actions/admin/appwizard/Security.groovy

```diff
@@ -167,6 +168,54 @@ public class Security extends AbstractPortofinoRealm {
         user
     }
 
+    AuthenticationInfo loadAuthenticationInfo(GoogleToken googleToken) {
+        boolean checkDomain = portofinoConfiguration.getProperty("google.check.domain")
+        String domain = portofinoConfiguration.getProperty("google.domain")
+        boolean createUserIfNotExists = portofinoConfiguration.getProperty("google.create.user.if.not.exists")
+
+        if (checkDomain) {
+            if (!googleToken.getHd().equals(domain)) {
+                logger.warn("Current google account '{}' is not in the domain '{}'", googleToken.email, domain)
+                throw new AuthenticationException("Login failed")
+            }
+        }
+
+        Session session = persistence.getSession(databaseName);
+        Criteria criteria = session.createCriteria(userTableEntityName);
+        criteria.add(Restrictions.eq(userNameProperty, googleToken.email));
+
+        List result = criteria.list()
+
+        if (result.size() == 1) {
+            def user = cleanUser(result[0])
+            return new SimpleAuthenticationInfo(user, "", name)
+        } else {
+            if (createUserIfNotExists) {
+                Map persistentUser = new HashMap()
+                persistentUser[userNameProperty] = googleToken.getPrincipal()
+                if(!StringUtils.isEmpty(userEmailProperty)) {
+                    persistentUser[userEmailProperty] = googleToken.getEmail()
+                }
+                // ADD ALL THE NOT-NULLABLE COLUMNS
+
+                def tx = session.transaction;
+                if (!tx.isActive())
+                    tx.begin()
+
+                try {
+                    session.save(userTableEntityName, (Object) persistentUser)
+                } catch (ConstraintViolationException e) {
+                    throw new ExistingUserException(e);
+                }
+                tx.commit()
+                return new SimpleAuthenticationInfo(persistentUser, "", name)
+            } else {
+                throw new IncorrectCredentialsException("Login failed");
+            }
+        }
+    }
+
+
```

Comment:
Attenzione: se portofino-google è un modulo opzionale, non possiamo creare un Security.groovy con il wizard che lo richiede.
Soluzione semplice: commentare questo codice e introdurre un'intestazione tipo
`//Please uncomment this to enable Google OAuth`

Soluzione più complessa: aggiungere un campo al wizard "enable Google OAuth" o simili e condizionare l'inserimento del codice a questo campo

Reply:
Ciao, per ora ho optato per la soluzione semplice, appena posso vorrei introdurre il campo nel wizard